### PR TITLE
HHH-18556 Expressions.nullExpresion() in querydsl result in NPE in SqmExpressible

### PR DESCRIPTION
**Pull request by dreab8**
https://hibernate.atlassian.net/browse/HHH-18556

https://github.com/hibernate/hibernate-orm/pull/8929 Backport 

----------------------
By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt)
and can be relicensed under the terms of the [LGPL v2.1 license](https://www.gnu.org/licenses/old-licenses/lgpl-2.1.txt) in the future at the maintainers' discretion.
For more information on licensing, please check [here](https://github.com/hibernate/hibernate-orm/blob/main/CONTRIBUTING.md#legal).

----------------------

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/query/spi/AbstractCommonQueryContract.java [hunk 1]
```diff
@@ -906,7 +907,7 @@ public CommonQueryContract setParameter(int position, Object value) {
 		final QueryParameter<Object> param = binding.getQueryParameter();
 		if ( param.allowsMultiValuedBinding() ) {
 			final BindableType<?> hibernateType = param.getHibernateType();
-			if ( hibernateType == null || isInstance( hibernateType, value ) ) {
+			if ( hibernateType == null || hibernateType instanceof NullSqmExpressible || isInstance( hibernateType, value ) ) {
```
Comment (gavinking) [inline hibernate-core/src/main/java/org/hibernate/query/spi/AbstractCommonQueryContract.java:910]: @dreab8 Why did you make this change for positional parameters, but not for named parameters?

Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/query/spi/AbstractCommonQueryContract.java:910]: Hi @gavinking, 

a big oversight :( I'm going to open a Jira and fix it.

Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/query/spi/AbstractCommonQueryContract.java:910]: thanks for spotting it

Comment (gavinking) [reply hibernate-core/src/main/java/org/hibernate/query/spi/AbstractCommonQueryContract.java:910]: Excellent, cheers mate 👍

**Pull request merged**